draft-age-masque-connect-ip.md

```diff
@@ -334,6 +343,12 @@ packets in that prefix to its peer. Note that this capsule can be reordered
 with DATAGRAM frames, and therefore an endpoint that receives packets for
 routes it has rejected MUST NOT treat that as an error.
 
+ROUTE_ADVERTISEMENT and ROUTE_WITHDRAWAL capsules are applied in order of
+receipt: if a prefix is covered by multiple received ROUTE_ADVERTISEMENT and/or
+ROUTE_WITHDRAWAL capsules, only the last received capsule applies as it
```

Comment:
This text is a bit confusing as written. Maybe rephrase this as "ROUTE_ADVERTISEMENT and ROUTE_WITHDRAWAL for a specific prefix [...]"?

Reply:
That would mean something different. I phrased this specifically to cover the scenario of one route containing another.

For example if we have the following sequence:
* ROUTE_ADVERTISEMENT 2001:DB8::/32
* ROUTE_WITHDRAWAL 2001:DB8:1111:2222/64

Then the recipient has routes in the tunnel for 2001:DB8:0000:0000-2001:DB8:1111:2221 and 2001:DB8:1111:2223-2001:DB8:FFFF:FFFF.

Reply:
In attempting to suggest text that I find more clear, I actually think I found a bug in these semantics.

We've overloaded WITHDRAWAL to mean two things in this setup:
1. Undo a previous ADVERTISEMENT
2. Create an entry in the routing table that means no-route

These are actually slightly different operations, even though it sounds like they should produce the same effect, because of the processing of overlap.

The in-order logic simply means whenever you get a new ADVERTISEMENT, you append it to your table -- pretty simple. But WITHDRAWAL has to first scan for an exact match in the table, and if it finds one, delete it. If it does not, then it needs to add an entry for no-route. But that implies the ADVERTISEMENT logic is similarly wrong: it has to also delete any no-route entries that may already exist.

Perhaps ADVERTISEMENT should grow another field in the capsule indicating if the range is routable or not, and WITHDRAW is simply an inverse of ADVERTISEMENT? That way a positive route and null route are represented without overload.

Reply:
In order to facilitate progress, I'm going to merge this as-is and continue the discussion in #34. That way I can write a PR based off main that can attempt to resolve #34 and we can discuss there.